src/main/java/com/icthh/xm/ms/entity/web/rest/FunctionResource.java

```diff
@@ -95,6 +95,26 @@ public void setSelf(@Lazy FunctionResource self) {
             .body(result.functionResult());
     }
 
+    /**
+     * POST  /functions/{functionKey} : Execute a function by key (key in entity specification).
+     *
+     * @param functionKey   the function key to execute
+     * @param functionInput function input data context
+     * @return the ResponseEntity with status 201 (Created) and with body the new FunctionContext,
+     * or with status 400 (Bad Request) if the FunctionContext has already an ID
+     */
+    @Timed
+    @PostMapping("/functions/anonymous/{functionKey:.+}")
+    @PreAuthorize("hasPermission({'functionKey': #functionKey}, 'FUNCTION.ANONYMOUS.CALL')")
```

Comment:
is it's ANONYMOUS function why permission exists?

Reply:
what difference between simple function

Reply:
pls remove this PreAuthorize

Reply:
removed